gr-uhd/lib/rfnoc_replay_impl.cc

```diff
@@ -0,0 +1,201 @@
+{
+    if (!(pmt::is_dict(msg)) && pmt::is_pair(msg)) {
+        d_logger->debug(
+            "Command message is pair, converting to dict: '{}': car({}), cdr({})",
+            pmt::write_string(msg),
+            pmt::write_string(pmt::car(msg)),
+            pmt::write_string(pmt::cdr(msg)));
+        msg = pmt::dict_add(pmt::make_dict(), pmt::car(msg), pmt::cdr(msg));
+    }
+
+    // Make sure, we use dicts!
+    if (!pmt::is_dict(msg)) {
+        d_logger->error("Command message is neither dict nor pair: {}",
+                        pmt::write_string(msg));
+        return;
+    }
+
+    if (!pmt::dict_has_key(msg, replay_cmd_key())) {
+        d_logger->error("Command message does not have a command: {}",
+                        pmt::write_string(msg));
+        return;
+    }
+
+    bool has_cmd_offset = false;
+    bool has_cmd_size = false;
+    uint64_t cmd_offset, cmd_size;
+
+    // Check to see if the command sent specifies a size and/or offset for the command
+    // If it does not that is fine, we will use the record/play size/offset properties.
+    if (pmt::dict_has_key(msg, replay_cmd_offset_key())) {
+        cmd_offset = int(
+            pmt::to_long(pmt::dict_ref(msg, replay_cmd_offset_key(), pmt::from_long(0))));
+        has_cmd_offset = true;
+    }
+    if (pmt::dict_has_key(msg, replay_cmd_size_key())) {
+        cmd_size = int(
+            pmt::to_long(pmt::dict_ref(msg, replay_cmd_size_key(), pmt::from_long(0))));
+        has_cmd_size = true;
+    }
+
+    int cmd_port = int(pmt::to_long(pmt::dict_ref(msg,
+                                                  replay_cmd_port_key(),
+                                                  pmt::from_long(0) // Default to Port 0
+                                                  )));
+
+    double cmd_time =
+        int(pmt::to_float(pmt::dict_ref(msg,
+                                        replay_cmd_time_key(),
+                                        pmt::from_float(0.0) // Default to 0.0
+                                        )));
+
+    std::string command =
+        pmt::symbol_to_string(pmt::dict_ref(msg, replay_cmd_key(), pmt::intern("")));
+
+    if (command == "record") {
+        // If command didn't specify size/offset get from the record properties
+        if (!has_cmd_offset)
+            cmd_offset = d_wrapped_ref->get_record_offset(cmd_port);
+        if (!has_cmd_size)
+            cmd_size = d_wrapped_ref->get_record_size(cmd_port);
+        record(cmd_offset, cmd_size, cmd_port);
+        return;
+    }
+    // If command didn't specify size/offset get from the play properties
+    if (!has_cmd_offset)
+        cmd_offset = d_wrapped_ref->get_play_offset(cmd_port);
+    if (!has_cmd_size)
+        cmd_size = d_wrapped_ref->get_play_size(cmd_port);
+    if (command == "play") {
+        play(cmd_offset, cmd_size, cmd_port, cmd_time, false);
+        return;
+    }
+    if (command == "play_continuous") {
```

Comment:
I'm thinking it might be more consistent if you make the `repeat` argument part of the command dict. Or you keep the `play_continuous` command, but allow the `play` command to carry a `repeat` argument which defaults to `false`. Or none of the above... I don't have a definitive opinion. Given your move to using a command dict, the `repeat` argument seems consistent though.